packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -46,6 +59,39 @@ export class ProblemWidget extends TreeWidget {
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
     }
 
+    @postConstruct()
+    protected init(): void {
+        super.init();
+        this.updateFollowActiveEditor();
+        this.toDispose.push(this.preferences.onPreferenceChanged(e => {
+            if (e.preferenceName === 'problems.autoReveal') {
+                this.updateFollowActiveEditor();
+            }
+        }));
+    }
+
+    protected updateFollowActiveEditor(): void {
+        this.toDisposeOnCurrentWidgetChanged.dispose();
+        if (this.preferences.get('problems.autoReveal')) {
+            this.followActiveEditor();
+            this.toDispose.push(this.toDisposeOnCurrentWidgetChanged);
+        }
+    }
+
+    protected followActiveEditor(): void {
+        this.toDisposeOnCurrentWidgetChanged.push(this.shell.onDidChangeCurrentWidget(e => {
```

Comment:
Should not we extract a listener in own method and call before as well? i.e. something:
```ts
protected followActiveEditor(): void {
   this.autoRevealFromActiveEditor();
   this.toDisposeOnCurrentWidgetChanged(() => this.autoRevealFromActiveEditor());
}
```

I wonder about the use case when `autoReveal` is enabled, the problems view is not visible and different editors get activated. When the problems view gets revealed it should automatically follow the active editor too.